packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,592 @@
+   */
+  @property({
+    attribute: 'drive-id'
+  })
+  public get driveId(): string {
+    return this._driveId;
+  }
+  public set driveId(value: string) {
+    if (value === this._driveId) {
+      return;
+    }
+
+    this._driveId = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide group id for a file
+   *
+   * @type {string}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'group-id'
+  })
+  public get groupId(): string {
+    return this._groupId;
+  }
+  public set groupId(value: string) {
+    if (value === this._groupId) {
+      return;
+    }
+
+    this._groupId = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide item id for a file
+   *
+   * @type {string}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'item-id'
+  })
+  public get itemId(): string {
+    return this._itemId;
+  }
+  public set itemId(value: string) {
+    if (value === this._itemId) {
+      return;
+    }
+
+    this._itemId = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide item path for a file
+   *
+   * @type {string}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'item-path'
+  })
+  public get itemPath(): string {
+    return this._itemPath;
+  }
+  public set itemPath(value: string) {
+    if (value === this._itemPath) {
+      return;
+    }
+
+    this._itemPath = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide user id for a file
+   *
+   * @type {string}
+   * @memberof MgtFile
+   */
+  @property({
+    attribute: 'user-id'
+  })
+  public get userId(): string {
+    return this._userId;
+  }
+  public set userId(value: string) {
+    if (value === this._userId) {
+      return;
+    }
+
+    this._userId = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * allows developer to provide insight type for a file
+   * can be trending, used, or shared
+   *
+   * @type {OfficeGraphInsightString}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'insight-type'
+  })
+  public get insightType(): OfficeGraphInsightString {
+    return this._insightType;
+  }
+  public set insightType(value: OfficeGraphInsightString) {
+    if (value === this._insightType) {
+      return;
+    }
+
+    this._insightType = value;
+    this.requestStateUpdate();
+  }
+
+  /**
+   * Sets what data to be rendered (file icon only, oneLine, twoLines threeLines).
+   * Default is 'threeLines'.
+   *
+   * @type {ViewType}
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'view',
+    converter: value => {
+      if (!value || value.length === 0) {
+        return ViewType.threelines;
+      }
+
+      value = value.toLowerCase();
+
+      if (typeof ViewType[value] === 'undefined') {
+        return ViewType.threelines;
+      } else {
+        return ViewType[value];
+      }
+    }
+  })
+  public view: ViewType;
```

Comment:
What do you think about `itemView` instead of `view` since it's setting the view of each item?

Reply:
Sure thing!